pkg/server/server_entrypoint_tcp.go

```diff
@@ -548,13 +570,57 @@ func createHTTPServer(ctx context.Context, ln net.Listener, configuration *stati
 		})
 	}
 
+	var lastHandler http.Handler
+	if debug || (configuration.Transport != nil && (configuration.Transport.KeepAliveMaxTime > 0 || configuration.Transport.KeepAliveMaxRequests > 0)) {
+		lastHandler = http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
Here I would do like above and have a dedicated unexposed function, to ease the readability (the function name is just an example) and have something like the following at the end:

```go
if configuration.Transport.KeepAliveMaxTime > 0 || configuration.Transport.KeepAliveMaxRequests > 0 {
handler = closeKeepAliveConnections(...)
}
```